modules/cnf-numa-resource-scheduling-strategies.adoc

```diff
@@ -0,0 +1,70 @@
+
+. Once a workload is scheduled, the selected NUMA node’s resources are updated to reflect the allocation.
+
+The default strategy applied is the `LeastAllocated` strategy. This assigns workloads to the NUMA node with the most available resources that is the least utilized NUMA node. The goal of this strategy is to spread workloads across NUMA nodes to reduce contention and avoid hotspots.
+
+.Example
+Assume a worker node has two NUMA nodes, and the workload requires 4 vCPUs and 8 GB of memory:
+
+[cols="5,2,2,2,2,2", options="header"]
+|===
+|NUMA Node |Total CPUs |Used CPUs |Total Memory (GB) |Used Memory (GB) |Available Resources
+|NUMA 1 |16 |12 |64 |56 |4 CPUs, 8 GB Memory
+|NUMA 2 |16 |6 |64 |24 |10 CPUs, 40 GB Memory
+|===
+
+NUMA 2 has more available resources (10 CPUs, 40 GB memory) compared to NUMA 1 (4 CPUs, 8 GB memory) and therefore with the default strategy in place the workload is assigned to NUMA 2.
+
+
+The `MostAllocated` strategy assigns workloads to the NUMA node with the least available resources that is the the most utilized NUMA node. The goal of this strategy is to consolidate workloads on fewer NUMA nodes, leaving other NUMA nodes idle or with free resources for energy efficiency or other workloads.
+
+.Example
+Using the same initial NUMA node states as earlier:
+
+[cols="2,2"]
+|===
+|NUMA Node |Available Resources
+|NUMA 1 |4 CPUs, 8 GB Memory
+|NUMA 2 |10 CPUs, 40 GB Memory
+|===
+
+NUMA 1 has fewer available resources compared to NUMA 2 and therefore with the `MostAllocated` strategy in place the workload is assigned to NUMA 1, further consolidating usage.
+
+The `BalancedAllocation` strategy assigns workloads to the NUMA node with the most balanced resource utilization across CPU and memory. The goal of this strategy is to avoid scenarios where one resource for example CPU is heavily utilized while another for example memory is underutilized.
+
+.Example
+Assume a worker node has the following NUMA node states:
+
+[cols="2,2,2,2"]
+|===
+|NUMA Node |CPU Usage |Memory Usage |BalancedAllocation Score
+|NUMA 1 |60% |55% |High (More Balanced)
+|NUMA 2 |80% |20% |Low (Less Balanced)
+|===
+
+NUMA 1 has a more balanced CPU and memory utilization compared to NUMA 2 and therefore with the `BalancedAllocation` strategy in place the workload is assigned to NUMA 1. 
+
+The following table summarizes the different strategies and their outcomes:
+
+[cols="2,3,3", options="header"]
+|===
+|Strategy |Description |Outcome
+|LeastAllocated |Favors NUMA nodes with the most available resources. |Spreads workloads to reduce contention and ensure headroom for high-priority tasks.
+|MostAllocated |Favors NUMA nodes with the least available resources. |Consolidates workloads on fewer NUMA nodes, freeing others for energy efficiency.
+|BalancedAllocation |Favors NUMA nodes with balanced CPU and memory usage. |Ensures even resource utilization, preventing skewed usage patterns.
```

Comment:
Can the strategy names be marked up? Also, seems like the descriptions could all lose punctuation since they're all fragments.

```suggestion
|`LeastAllocated` |Favors NUMA nodes with the most available resources |Spreads workloads to reduce contention and ensure headroom for high-priority tasks
|`MostAllocated` |Favors NUMA nodes with the least available resources |Consolidates workloads on fewer NUMA nodes, freeing others for energy efficiency
|`BalancedAllocation` |Favors NUMA nodes with balanced CPU and memory usage |Ensures even resource utilization, preventing skewed usage patterns
```